plugins/guests/darwin/cap/change_host_name.rb

```diff
@@ -6,7 +6,7 @@ def self.change_host_name(machine, name)
           if !machine.communicate.test("hostname -f | grep '^#{name}$' || hostname -s | grep '^#{name}$'")
             machine.communicate.sudo("scutil --set ComputerName #{name}")
             machine.communicate.sudo("scutil --set HostName #{name}")
-            machine.communicate.sudo("scutil --set LocalHostName #{name}")
+            machine.communicate.sudo("scutil --set LocalHostName #{name.split('.')[0]}")
```

Comment:
@sethvargo It might be reasonable, but I'd like to notice that `LocalHostName` is not a hostname itself. It's just the name identifier used by Bonjour and visible through file sharing services like AirDrop. 
Do you think that we should warn the user anyway?

In GUI this parameter can be set there: "System Preferences" -> "Sharing" -> "Edit..."
I've noticed that OS X suggests `LocalHostName` value based on `ComputerName` with removed dots. May be we should follow the same approach here, in vagrant...
![sharing](https://cloud.githubusercontent.com/assets/4846154/7746669/fdeec244-ffbc-11e4-80c1-bb0a22887953.png)
(".local" is added automatically)